language/stdlib/modules/association.move

```diff
@@ -109,6 +114,156 @@ module Association {
     acquires PrivilegedCapability {
         Transaction::assert(addr_is_association(addr), 1002);
     }
-}
 
+// **************** SPECIFICATIONS ****************
+// Note: I experimented with using Markdown to organize this, but I haven't converted
+// it to work with the new doc generator, yet.
+
+// # Specifications for association.move
+
+// This is preliminary.  This is one of the first real module libraries with global
+// specifications, and everything is evolving.
+
+    spec module {
+        pragma verify = true;
+
+        // This mirrors the Move function root_address()
+        define spec_root_address(): address { 0xA550C18 }
+
+        // mirrors Move addr_is_association
+        define spec_addr_is_association(addr: address): bool {
+            exists<PrivilegedCapability<T>>(addr)
+            && global<PrivilegedCapability<T>>(addr).is_certified
+        }
+     }
+
+    // ## Management of privileges
+    // TODO: With grant/remove etc., separately specify that only root may grant or remove privileges
+
+    // ### Management of Root {} marker.
+
+    // The root_address is marked by a Root{} object that is stored at that place only.
+    spec module {
+        // enclose this in a function so we can easily repeat the same property in the invariant
+        // and as a post-condition to initialize (the base case).
+        // SPEC: Only the root address has a Root{} resource.
```

Comment:
I'm trying to mark English versions of the formal specs.  I was experimenting with a convention where every logical spec is also written in English (separately from other explanation, etc.)

Does that seem like a good idea?

Reply:
I changed SPEC: to "Informally:".  I think it would be a good idea to restate every property in English.  It makes it more readable, and, if someone spots a discrepancy between the English and the spec language, it might mean there's an error in the formal spec.